test/unit/npm/publisher.test.js

```diff
@@ -77,6 +83,56 @@ describe('npm/publisher.js', function () {
     assume(req.headers.host).equals(parsed.host);
   });
 
+  describe('build', function () {
+    const uri = actualConfig.builder.url;
+
+    after(function () {
+      sinon.restore();
+    });
+
+    it('succeeds to publish after retry', function (done) {
```

Comment:
do we have tests covering success and outright failures?

Reply:
@msluther we didn't do that intentionally since testing stream here is a bit weird which also need to think hard about timing (event loop specific stuff). @jcrugzz found this work around to make it work - fail at the first and then succeed with retry.